### PR TITLE
Module5 homework task

### PR DESCRIPTION
**Pull request by dmnikitin**
All evaluation criteria points were met, including extra task.

### CODE REVIEW

File: src/app/core/services/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  private userName: string;
+  private accessToken: string;
+  constructor(private router: Router) { }
+
+  public isAuthenticated(): boolean {
+    const userName: string = localStorage.getItem('userName');
+    const accessToken: string = localStorage.getItem('accessToken');
+    if (userName && accessToken) {
+      this.authencticationStatus = true;
+      this.userName = userName;
```
Comment (KonanMentor) [inline src/app/core/services/auth.service.ts:19]: In general it's a bad practice to do side effects in getters

File: src/app/core/services/courses.service.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  public getItemById(id: string): Course {
+    return this.courses.find((course)=> course.id === id );
+  }
+
+  public updateItem(id: string, newData: Course): void {
+    this.courses = this.courses.map((item) => {
+      if (item.id === id) {
+        return newData;
+      }
```
Comment (KonanMentor) [inline src/app/core/services/courses.service.ts:34]: It will replace all other items with `undefined` :slightly_smiling_face:

**Review by KonanMentor — COMMENTED** [review]
:+1: for unit tests

File: src/app/core/services/courses.service.ts (hunk 2 above)
Comment (dmnikitin) [reply src/app/core/services/courses.service.ts:34]: my bad, forgot to add else statement.

**Pull request merged**